synapse/util/metrics.py

```diff
@@ -57,8 +58,10 @@
     sub_metrics=["real_time_max", "real_time_sum"],
 )
 
+T = TypeVar("T", bound=Callable[..., Any])
 
-def measure_func(name=None):
+
+def measure_func(name: str = None) -> Callable[[T], T]:
```

Comment:
We seem to usually be explicit about optional parameters.

```suggestion
def measure_func(name: Optional[str] = None) -> Callable[[T], T]:
```